v2/pkg/engine/datasource/graphql_datasource/graphql_datasource.go

```diff
@@ -1944,6 +1946,15 @@ func (s *SubscriptionSource) AsyncStart(ctx *resolve.Context, id uint64, input [
 	if options.Body.Query == "" {
 		return resolve.ErrUnableToResolve
 	}
+	for _, fn := range s.onSubscriptionStartFns {
+		events, err := fn(ctx)
+		if err != nil {
+			return err
+		}
+		for _, event := range events {
+			updater.Update(event)
```

Comment:
Is it the right approach to call the updater.Update func or should we just handle the event directly?